server/controllers/finance/priceList.js

```diff
@@ -0,0 +1,317 @@
+    return [
+      item.uuid || uuid(),
+      item.inventory_uuid,
+      priceListUuid,
+      item.label,
+      item.value,
+      item.is_percentage
+    ];
+  });
+}
+
+
+/**
+ * Create a new price list entity, along with price list items.
+ *
+ * POST /prices
+ */
+exports.create = function create(req, res, next) {
+  'use strict';
+
+  var items;
+  var data = req.body.list;
+  var trans = db.transaction();
+  var priceListSql =
+    'INSERT INTO price_list (uuid, label, description, enterprise_id) ' +
+    'VALUES (?, ?, ?, ?);';
+  var priceListItemSql =
+    'INSERT INTO price_list_item (uuid, inventory_uuid, price_list_uuid, ' +
+    'label, value, is_percentage) VALUES ?;';
+
+
+  // generate a UUID if not provided
+  data.uuid = data.uuid || uuid();
+
+  // if the client didn't send price list items, do not create them.
+  if (data.items) {
+    items = formatPriceListItems(data.uuid, data.items);
+  }
+
+  // remove before database insertion
+  delete data.items;
+
+  // enqueue the queries to the transaction
+  trans.addQuery(priceListSql, [
+    data.uuid, data.label, data.description, req.session.enterprise.id
+  ]);
+
+  // only enqueue the price list items if they exist
+  if (items) { trans.addQuery(priceListItemSql, [ items ]); }
+
+  trans.execute()
+  .then(function () {
+
+    // respond to the client with a 201 CREATED
+    res.status(201).json({
+      uuid : data.uuid
+    });
+  })
+  .catch(next)
+  .done();
+};
+
+
+/**
+ * Updates a price list (and associated items) in the database.
+ *
+ * PUT /prices/:uuid
+ */
+exports.update = function update(req, res, next) {
+  'use strict';
+
+  var items;
+  var data = req.body.list;
+  var priceListSql =
+    'UPDATE price_list SET ? WHERE uuid = ?;';
+
+  var priceListDeleteItemSql =
+    'DELETE FROM price_list_item WHERE price_list_uuid = ?';
+
+  var priceListCreateItemSql =
+    'INSERT INTO price_list_item (uuid, inventory_uuid, price_list_uuid, ' +
+    'label, value, is_percentage) VALUES ?;';
+
+  var trans = db.transaction();
+
+  // if the client didn't send price list items, do not create them.
+  if (data.items) {
+    items = formatPriceListItems(req.params.uuid, data.items);
+  }
+
+  // remove non-updatable properties before queries
+  delete data.uuid;
+  delete data.items;
+
+  // make sure the price list exists
+  lookupPriceList(req.params.uuid)
+  .then(function (priceList) {
+
+    // if we get here, it means the price list exists and we can perform an
+    // update query on it. Since we are doing multiple operations at once,
+    // wrap the queries in a DB transaction.
+    if (!isEmptyObject(data)) {
+      trans.addQuery(priceListSql, [ data, req.params.uuid ]);
+    }
+
+    // only trigger price list item updates if the items have been sent back to
+    // the server
+    if (items) {
+      trans.addQuery(priceListCreateItemSql, [ items ]);
+      trans.addQuery(priceListDeleteItemSql, [ req.params.uuid ]);
```

Comment:
This has been mentioned but addQuery returns the base object allowing you to chain the function. 

``` js
trans
   .addQuery(sql, [])
   .addQuery(sql, []);
```